### DISCUSSION TITLE
How to solve duplicate vertex in local convex hull calculation

### DISCUSSION THREAD

**Discussion by Jeannis0** [jmsigner/amt#110] [Q&A]
Hello and thank you for this great package! 

I am currently working on GPS data which I turned into a track and then nested it (based on the workflow of the paper) and am now stumbling upon an error message once I want to use the `hr_locoh` function on my nested track data.

```
Caused by error in `wk_handle.wk_wkb()`:
! Loop 2 is not valid: Edge 1446 is degenerate (duplicate vertex)
```

As I understand this is an error related to the `sf` package which is used in the `hr_locoh` function.
The usual workarounds for this error, that I found so far are:
`st_make_valid()` - which seems to not work, as the error is caused deeper within the hr_locoh function
and
`sf_use_s2(FALSE)` - which I'd rather not use, as it changes the entire formula for the spatial sf computation.

Do you have any workaround/solution to this issue?

**Comment by jmsigner** [comment]
Likely, we do not catch a invalid geometry. Could share a small data set that reproduces the error (by email is fine).
